rai/client.go

```diff
@@ -353,14 +353,19 @@ func readArrowFiles(files []TransactionAsyncFile) ([]ArrowRelation, error) {
 					data, _ := rec.Column(i).MarshalJSON()
 					var values []interface{}
 					json.Unmarshal(data, &values)
-					out = append(out, ArrowRelation{file.Name, values})
+					res[file.Name] = append(res[file.Name], values)
 				}
 
 				rec.Retain()
 			}
 		}
 	}
 
+	var out []ArrowRelation
+	for k, v := range res {
+		out = append(out, ArrowRelation{k, v})
+	}
+
```

Comment:
Now that I understand, I think this could still be done in a single pass. I think something like this should work:
https://github.com/RelationalAI/rai-sdk-go/pull/22/

Can you have a look? Thanks!!